File: manager/app/src/main/java/com/anatdx/yukisu/magica/BootCompletedReceiver.java
package com.anatdx.yukisu.magica;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootCompletedReceiver extends BroadcastReceiver {
    public static final String ACTION_LAUNCH = "com.anatdx.yukisu.magica.LAUNCH";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null) {
            return;
        }

        final String action = intent.getAction();
        if (!Intent.ACTION_LOCKED_BOOT_COMPLETED.equals(action)
                && !Intent.ACTION_BOOT_COMPLETED.equals(action)
                && !ACTION_LAUNCH.equals(action)) {
            return;
        }

        MagicaHelper.launch(context, action);
    }
}

File: manager/app/src/main/java/com/anatdx/yukisu/magica/BootCompletedReceiver.kt
package com.anatdx.yukisu.magica

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.anatdx.yukisu.ui.util.rootAvailable

open class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        val action = intent?.action ?: return
        if (
            action != Intent.ACTION_LOCKED_BOOT_COMPLETED &&
            action != Intent.ACTION_BOOT_COMPLETED &&
            action != ACTION_LAUNCH
        ) {
            return
        }
        if (rootAvailable()) {
            return
        }

        MagicaHelper.launch(context, action)
    }

    companion object {
        const val ACTION_LAUNCH = "com.anatdx.yukisu.magica.LAUNCH"
    }
}
